drivers/soundwire/intel.c

```diff
@@ -423,7 +426,9 @@ static int intel_link_power_down(struct sdw_intel *sdw)
 	cpa_mask = (SDW_SHIM_LCTL_CPA << link_id);
 	link_control &=  spa_mask;
 
+	mutex_lock(&sdw->link_power_lock);
 	ret = intel_clear_bit(shim, SDW_SHIM_LCTL, link_control, cpa_mask);
+	mutex_unlock(&sdw->link_power_lock);
```

Comment:
intel_link_power_down() is only called in suspend routines.

intel_link_power_up() is called on startup and resume routines

How can we have a conflict here?